python/lammps/mliap/pytorch.py

```diff
@@ -132,22 +133,26 @@ def forward(self, elems, descriptors, beta, energy,use_gpu_data=False):
         descriptors = torch.as_tensor(descriptors,dtype=self.dtype, device=self.device).requires_grad_(True)
         elems = torch.as_tensor(elems,dtype=torch.int32, device=self.device)
         elems=elems-1
+        device = self.device
+        if (use_gpu_data and device == None and str(beta.device).find('CUDA') == 1):
+            device = 'cuda' #Override device as it wasn't defined in the model
         with torch.autograd.enable_grad():
 
             if (use_gpu_data):
-                energy_nn = torch.as_tensor(energy,dtype=self.dtype, device=self.device)
+                energy_nn = torch.as_tensor(energy,dtype=self.dtype, device=device)
                 energy_nn[:] = self.model(descriptors, elems).flatten()
             else:
                 energy_nn = self.model(descriptors, elems).flatten()
                 energy[:] = energy_nn.detach().cpu().numpy().astype(np.float64)
-
+                
```

Comment:
Needs whitespace cleanup (remove trailing spaces) on line 147

Reply:
Just fixed this